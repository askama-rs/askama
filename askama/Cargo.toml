[package]
name = "askama"
version = "0.13.0"
description = "Type-safe, compiled Jinja-like templates for Rust"
keywords = ["markup", "template", "jinja2", "html"]
categories = ["template-engine"]
homepage = "https://askama.readthedocs.io/"
repository = "https://github.com/askama-rs/askama"
license = "MIT OR Apache-2.0"
edition = "2021"
rust-version = "1.81"

[package.metadata.docs.rs]
features = ["full"]
rustdoc-args = ["--generate-link-to-definition", "--cfg=docsrs"]

[[bench]]
name = "to-json"
harness = false
required-features = ["serde_json"]

[[bench]]
name = "escape"
harness = false

[dependencies]
itoa = "1.0.11"

# needed by feature "derive"
askama_derive = { version = "=0.13.0", path = "../askama_derive", default-features = false, optional = true }

# needed by feature "serde_json"
serde = { version = "1.0", optional = true, default-features = false }
serde_json = { version = "1.0", optional = true, default-features = false }

# needed by feature "urlencode"
percent-encoding = { version = "2.1.0", optional = true, default-features = false }

[dev-dependencies]
assert_matches = "1.5.0"
criterion = "0.5"

[badges]
maintenance = { status = "actively-developed" }

[features]
default = ["config", "derive", "std", "urlencode", "askama_derive?/default"]
full = ["default", "blocks", "code-in-doc", "serde_json", "askama_derive?/full"]

alloc = [
    "askama_derive?/alloc",
    "serde?/alloc",
    "serde_json?/alloc",
    "percent-encoding?/alloc",
]
blocks = ["askama_derive?/blocks"]
code-in-doc = ["askama_derive?/code-in-doc"]
config = ["askama_derive?/config"]
derive = ["askama_derive"]
serde_json = ["std", "askama_derive?/serde_json", "dep:serde", "dep:serde_json"]
std = [
    "alloc",
    "askama_derive?/std",
    "serde?/std",
    "serde_json?/std",
    "percent-encoding?/std",
]
urlencode = ["askama_derive?/urlencode", "dep:percent-encoding"]
