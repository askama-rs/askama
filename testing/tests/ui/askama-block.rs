use askama::Template;

#[derive(Template)]
#[template(ext = "txt", in_doc = true)]
/// Some documentation
///
/// ```html,askama
/// {% if true %}
///     {% fail %}
/// {% endif %}
/// ```
struct SyntaxError;

#[derive(Template)]
#[template(ext = "txt", in_doc = true)]
/// `````
/// ```askama
/// {{bla}}
/// ```
/// `````
struct BlockInBlock;

#[derive(Template)]
#[template(ext = "txt")]
/// Some documentation
///
/// ```html,askama
/// Hello.
/// ```
struct InDocMissing;

#[derive(Template)]
#[template(ext = "txt", in_doc)]
/// Some documentation
///
/// ```html,askama
/// Hello.
/// ```
struct InDocEmpty;

#[derive(Template)]
#[template(ext = "txt", in_doc = false)]
/// Some documentation
///
/// ```html,askama
/// Hello.
/// ```
struct InDocWrong;

#[derive(Template)]
#[template(ext = "txt", in_doc = "yes")]
/// Some documentation
///
/// ```html,askama
/// Hello.
/// ```
struct InDocWrongType;

#[derive(Template)]
#[template(ext = "txt", in_doc = true)]
enum NoDocForEnum {
    Yes,
    No,
}

#[derive(Template)]
#[template(ext = "txt", in_doc = true)]
union NoDocForUnion {
    u: u8,
    i: i8,
}

fn main() {}
