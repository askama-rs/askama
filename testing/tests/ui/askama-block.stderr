error: unknown node `fail`
 --> <source attribute>:2:6
       " fail %}\n{% endif %}"
 --> tests/ui/askama-block.rs:5:1
  |
5 | /// Some documentation
  | ^^^^^^^^^^^^^^^^^^^^^^

error: when using `in_doc` with the value `true`, the struct's documentation needs a `askama` code block
  --> tests/ui/askama-block.rs:15:34
   |
15 | #[template(ext = "txt", in_doc = true)]
   |                                  ^^^^

error: specify one template argument `path`, `source` or `in_doc`
  --> tests/ui/askama-block.rs:24:3
   |
24 | #[template(ext = "txt")]
   |   ^^^^^^^^

error: unsupported attribute argument
  --> tests/ui/askama-block.rs:33:25
   |
33 | #[template(ext = "txt", in_doc)]
   |                         ^^^^^^

error: specify one template argument `path`, `source` or `in_doc`
  --> tests/ui/askama-block.rs:42:3
   |
42 | #[template(ext = "txt", in_doc = false)]
   |   ^^^^^^^^

error: template attribute `in_doc` expects a boolean value
  --> tests/ui/askama-block.rs:51:34
   |
51 | #[template(ext = "txt", in_doc = "yes")]
   |                                  ^^^^^

error: when using `in_doc` with the value `true`, the enum's documentation needs a `askama` code block
  --> tests/ui/askama-block.rs:60:34
   |
60 | #[template(ext = "txt", in_doc = true)]
   |                                  ^^^^

error: askama templates are not supported for `union` types, only `struct` and `enum`
  --> tests/ui/askama-block.rs:68:1
   |
68 | union NoDocForUnion {
   | ^^^^^
